docs/database/dynamodb.md

```diff
@@ -83,6 +83,7 @@ These attributes are filled in for all DynamoDB request types.
 | `sybase` | Sybase | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `teradata` | Teradata | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 | `trino` | Trino | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `vector` | vector | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
Many of the databases listed here support vector search. E.g. mongoDB, cosmosDB, cassandra and probably others. A lot of databases are adding vector search due to it's high demand in AI world. Only one `db.system` can be specified on a telemetry item.

So, let's avoid adding a DB feature as a system. Instead, let's define attributes specific to vector search that could be used on any database that supports them.

Reply:
This means you are in favor to have a generic `vector` attribute to capture a generic vector search not specific to any vendor? This was the goal of this PR, that's why I proposed only few attributes after checking some vendors. But I think the `other_vector` proposed by @gregkalapos makes more sense if the vendor is not listed. The goal is the same.

Reply:
do we have some real-life case of an instrumentation that does not know the product name of the database?
If so, `other_vector` would make sense.
`other_sql` exists because of JDBC/ODBC/etc drivers and ORMs which might not know what database is used underneath.

If specific databases we need are not listed here, let's just list them and use their names.

Reply:
There are some proposals for ORM with vector database and some other framework like [Spring AI](https://docs.spring.io/spring-ai/reference/api/vectordbs.html) or [Semantic Kernel](https://learn.microsoft.com/en-us/semantic-kernel/integrations/#available-connectors-to-vector-databases) that are trying to achieve this. But right now I tend to agree, there are no many real-life case.